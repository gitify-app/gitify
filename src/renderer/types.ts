import type { OcticonProps } from '@primer/octicons-react';
import type { FC } from 'react';
import type { Notification, Reason } from './typesGitHub';
import type {
  AuthMethod,
  EnterpriseAccount,
  PlatformType,
} from './utils/auth/types';

declare const __brand: unique symbol;

type Brand<B> = { [__brand]: B };

export interface AuthState {
  accounts: Account[];
  /**
   * @deprecated This attribute is deprecated and will be removed in a future release.
   */
  token?: Token;
  /**
   * @deprecated This attribute is deprecated and will be removed in a future release.
   */
  enterpriseAccounts?: EnterpriseAccount[];
  /**
   * @deprecated This attribute is deprecated and will be removed in a future release.
   */
  user?: GitifyUser | null;
}

export type Branded<T, B> = T & Brand<B>;

export type AuthCode = Branded<string, 'AuthCode'>;

export type Token = Branded<string, 'Token'>;

export type ClientID = Branded<string, 'ClientID'>;

export type ClientSecret = Branded<string, 'ClientSecret'>;

export type Hostname = Branded<string, 'Hostname'>;

export type Link = Branded<string, 'WebUrl'>;

export type Status = 'loading' | 'success' | 'error';

export interface Account {
  method: AuthMethod;
  platform: PlatformType;
  version?: string;
  hostname: Hostname;
  token: Token;
  user: GitifyUser | null;
  hasRequiredScopes?: boolean;
}

export type SettingsValue =
  | boolean
  | number
  | GroupBy
  | OpenPreference
  | Reason[]
  | Theme;

export type SettingsState = AppearanceSettingsState &
  NotificationSettingsState &
  SystemSettingsState &
  FilterSettingsState;

export interface AppearanceSettingsState {
  theme: Theme;
  zoomPercentage: number;
  detailedNotifications: boolean;
  showAccountHeader: boolean;
  showPills: boolean;
  showNumber: boolean;
  wrapNotificationTitle: boolean;
}

export interface NotificationSettingsState {
  groupBy: GroupBy;
  fetchAllNotifications: boolean;
  participating: boolean;
  markAsDoneOnOpen: boolean;
  markAsDoneOnUnsubscribe: boolean;
  delayNotificationState: boolean;
}

export interface SystemSettingsState {
  openLinks: OpenPreference;
  keyboardShortcut: boolean;
  showNotificationsCountInTray: boolean;
  showNotifications: boolean;
  useAlternateIdleIcon: boolean;
  playSound: boolean;
  openAtStartup: boolean;
}

export interface FilterSettingsState {
  hideBots: boolean;
  filterReasons: Reason[];
}

export interface GitifyState {
  auth?: AuthState;
  settings?: SettingsState;
}

export enum Theme {
  SYSTEM = 'SYSTEM',
  LIGHT = 'LIGHT',
  LIGHT_HIGH_CONTRAST = 'LIGHT_HIGH_CONTRAST',
  LIGHT_COLORBLIND = 'LIGHT_COLORBLIND',
  LIGHT_TRITANOPIA = 'LIGHT_TRITANOPIA',
  DARK = 'DARK',
  DARK_HIGH_CONTRAST = 'DARK_HIGH_CONTRAST',
  DARK_COLORBLIND = 'DARK_COLORBLIND',
  DARK_TRITANOPIA = 'DARK_TRITANOPIA',
  DARK_DIMMED = 'DARK_DIMMED',
}

export enum OpenPreference {
  FOREGROUND = 'FOREGROUND',
  BACKGROUND = 'BACKGROUND',
}

export enum GroupBy {
  REPOSITORY = 'REPOSITORY',
  DATE = 'DATE',
}

export type RadioGroupItem = {
  label: string;
  value: string;
};

export interface AccountNotifications {
  account: Account;
  notifications: Notification[];
  error: GitifyError | null;
}

export interface GitifyUser {
  login: string;
  name: string | null;
  avatar: Link | null;
  id: number;
}

export interface GitifyError {
  title: string;
  descriptions: string[];
  emojis: string[];
}

export type ErrorType =
  | 'BAD_CREDENTIALS'
  | 'MISSING_SCOPES'
  | 'NETWORK'
  | 'RATE_LIMITED'
  | 'UNKNOWN';

export interface FormattedReason {
  title: string;
  description: string;
}

export enum IconColor {
  GRAY = 'text-gitify-icon-muted',
  GREEN = 'text-gitify-icon-open',
  PURPLE = 'text-gitify-icon-done',
  RED = 'text-gitify-icon-closed',
  YELLOW = 'text-gitify-icon-attention',
}

export enum Opacity {
  READ = 'opacity-50',
  LOW = 'opacity-70',
  MEDIUM = 'opacity-80',
  HIGH = 'opacity-90',
}

export type PullRequestApprovalIcon = {
  type: FC<OcticonProps>;
  color: IconColor;
  description: string;
};

export enum Size {
  XSMALL = 12,
  SMALL = 14,
  MEDIUM = 16,
  LARGE = 18,
  XLARGE = 20,
}

export type Chevron = {
  icon: FC<OcticonProps>;
  label: string;
};
