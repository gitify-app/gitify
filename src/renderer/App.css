html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

html.dark::-webkit-scrollbar-track,
html.dark div.flex-grow::-webkit-scrollbar-track {
  background-color: #090e15;
}

html.dark::-webkit-scrollbar-thumb,
html.dark div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #24292e;
}

html.dark::-webkit-scrollbar-thumb:hover,
html.dark div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #3a3f44;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

#nprogress {
  pointer-events: none;
}

#nprogress .bar {
  position: fixed;
  top: 0;
  left: 50px;
  z-index: 1031;
  background: #203354;

  width: 100%;
  height: 2px;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
