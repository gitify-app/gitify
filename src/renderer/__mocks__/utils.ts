import * as helpers from '../utils/helpers';

export function mockDirectoryPath() {
  jest.spyOn(helpers, 'getDirectoryPath').mockReturnValue('/mocked/dir/name');
}

/**
 * Ensure stable snapshots for our randomized emoji use-cases
 */
export function ensureStableEmojis() {
  global.Math.random = jest.fn(() => 0.1);
}
