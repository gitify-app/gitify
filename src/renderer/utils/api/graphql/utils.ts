export function formatAsGitHubSearchSyntax(
  repo: string,
  title: string,
): string {
  return `${title} in:title repo:${repo}`;
}
