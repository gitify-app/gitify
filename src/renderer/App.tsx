import { useContext } from 'react';
import {
  Navigate,
  Route,
  HashRouter as Router,
  Routes,
  useLocation,
} from 'react-router-dom';

import { BaseStyles, Box, ThemeProvider } from '@primer/react';

import { Loading } from './components/Loading';
import { Sidebar } from './components/Sidebar';
import { AppContext, AppProvider } from './context/App';
import { AccountsRoute } from './routes/Accounts';
import { FiltersRoute } from './routes/Filters';
import { LoginRoute } from './routes/Login';
import { LoginWithOAuthAppRoute } from './routes/LoginWithOAuthApp';
import { LoginWithPersonalAccessTokenRoute } from './routes/LoginWithPersonalAccessToken';
import { NotificationsRoute } from './routes/Notifications';
import { SettingsRoute } from './routes/Settings';

import './App.css';

function RequireAuth({ children }) {
  const { isLoggedIn } = useContext(AppContext);
  const location = useLocation();

  return isLoggedIn ? (
    children
  ) : (
    <Navigate to="/login" replace state={{ from: location }} />
  );
}

export const App = () => {
  return (
    <ThemeProvider>
      <BaseStyles>
        <AppProvider>
          <Router>
            <Box className="flex flex-col min-h-screen overflow-x-hidden overflow-y-auto pl-sidebar bg-gitify-background">
              <Loading />
              <Sidebar />
              <Routes>
                <Route
                  path="/"
                  element={
                    <RequireAuth>
                      <NotificationsRoute />
                    </RequireAuth>
                  }
                />
                <Route
                  path="/filters"
                  element={
                    <RequireAuth>
                      <FiltersRoute />
                    </RequireAuth>
                  }
                />
                <Route
                  path="/settings"
                  element={
                    <RequireAuth>
                      <SettingsRoute />
                    </RequireAuth>
                  }
                />
                <Route
                  path="/accounts"
                  element={
                    <RequireAuth>
                      <AccountsRoute />
                    </RequireAuth>
                  }
                />
                <Route path="/login" element={<LoginRoute />} />
                <Route
                  path="/login-personal-access-token"
                  element={<LoginWithPersonalAccessTokenRoute />}
                />
                <Route
                  path="/login-oauth-app"
                  element={<LoginWithOAuthAppRoute />}
                />
              </Routes>
            </Box>
          </Router>
        </AppProvider>
      </BaseStyles>
    </ThemeProvider>
  );
};
