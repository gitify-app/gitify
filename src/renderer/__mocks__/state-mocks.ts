import {
  type Account,
  type AuthState,
  type GitifyState,
  type GitifyUser,
  GroupBy,
  type Hostname,
  type Link,
  OpenPreference,
  type SettingsState,
  Theme,
  type Token,
} from '../types';
import type { EnterpriseAccount } from '../utils/auth/types';
import { Constants } from '../utils/constants';

export const mockEnterpriseAccounts: EnterpriseAccount[] = [
  {
    hostname: 'github.gitify.io' as Hostname,
    token: '1234568790' as Token,
  },
];

export const mockGitifyUser: GitifyUser = {
  login: 'octocat',
  name: 'Mona Lisa Octocat',
  id: 123456789,
  avatar: 'https://avatars.githubusercontent.com/u/583231?v=4' as Link,
};

export const mockPersonalAccessTokenAccount: Account = {
  platform: 'GitHub Cloud',
  method: 'Personal Access Token',
  token: 'token-123-456' as Token,
  hostname: Constants.DEFAULT_AUTH_OPTIONS.hostname,
  user: mockGitifyUser,
  hasRequiredScopes: true,
};

export const mockOAuthAccount: Account = {
  platform: 'GitHub Enterprise Server',
  method: 'OAuth App',
  token: '1234568790' as Token,
  hostname: 'github.gitify.io' as Hostname,
  user: mockGitifyUser,
  hasRequiredScopes: true,
};

export const mockGitHubCloudAccount: Account = {
  platform: 'GitHub Cloud',
  method: 'Personal Access Token',
  token: 'token-123-456' as Token,
  hostname: Constants.DEFAULT_AUTH_OPTIONS.hostname,
  user: mockGitifyUser,
  version: 'latest',
  hasRequiredScopes: true,
};

export const mockGitHubEnterpriseServerAccount: Account = {
  platform: 'GitHub Enterprise Server',
  method: 'Personal Access Token',
  token: '1234568790' as Token,
  hostname: 'github.gitify.io' as Hostname,
  user: mockGitifyUser,
  hasRequiredScopes: true,
};

export const mockGitHubAppAccount: Account = {
  platform: 'GitHub Cloud',
  method: 'GitHub App',
  token: '987654321' as Token,
  hostname: Constants.DEFAULT_AUTH_OPTIONS.hostname,
  user: mockGitifyUser,
  hasRequiredScopes: true,
};

export const mockAuth: AuthState = {
  accounts: [mockGitHubCloudAccount, mockGitHubEnterpriseServerAccount],
};

export const mockToken = 'token-123-456' as Token;

const mockAppearanceSettings = {
  theme: Theme.SYSTEM,
  zoomPercentage: 100,
  detailedNotifications: true,
  showPills: true,
  showNumber: true,
  showAccountHeader: false,
};

const mockNotificationSettings = {
  groupBy: GroupBy.REPOSITORY,
  fetchAllNotifications: true,
  participating: false,
  markAsDoneOnOpen: false,
  markAsDoneOnUnsubscribe: false,
  delayNotificationState: false,
};

const mockSystemSettings = {
  openLinks: OpenPreference.FOREGROUND,
  keyboardShortcut: true,
  showNotificationsCountInTray: true,
  showNotifications: true,
  playSound: true,
  useAlternateIdleIcon: false,
  openAtStartup: false,
};

const mockFilters = {
  hideBots: false,
  filterReasons: [],
};

export const mockSettings: SettingsState = {
  ...mockAppearanceSettings,
  ...mockNotificationSettings,
  ...mockSystemSettings,
  ...mockFilters,
};

export const mockState: GitifyState = {
  auth: mockAuth,
  settings: mockSettings,
};
