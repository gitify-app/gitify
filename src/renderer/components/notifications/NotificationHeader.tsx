import { type FC, type MouseEvent, useContext } from 'react';

import { Box, Stack } from '@primer/react';

import { AppContext } from '../../context/App';
import { GroupBy, Opacity, Size } from '../../types';
import type { Notification } from '../../typesGitHub';
import { cn } from '../../utils/cn';
import { openRepository } from '../../utils/links';
import { AvatarWithFallback } from '../avatars/AvatarWithFallback';

interface INotificationHeader {
  notification: Notification;
}

export const NotificationHeader: FC<INotificationHeader> = ({
  notification,
}: INotificationHeader) => {
  const { settings } = useContext(AppContext);

  const repoSlug = notification.repository.full_name;

  const notificationNumber = notification.subject?.number
    ? `#${notification.subject.number}`
    : '';

  const groupByDate = settings.groupBy === GroupBy.DATE;

  return (
    groupByDate && (
      <Box className="py-0.5">
        <Stack direction="horizontal" align="center" gap="condensed">
          <Box
            title="Open repository"
            className="text-xs font-medium"
            onClick={(event: MouseEvent<HTMLElement>) => {
              // Don't trigger onClick of parent element.
              event.stopPropagation();
              openRepository(notification.repository);
            }}
            data-testid="view-repository"
          >
            <AvatarWithFallback
              src={notification.repository.owner.avatar_url}
              alt={repoSlug}
              name={repoSlug}
              size={Size.SMALL}
              userType={notification.repository.owner.type}
            />
          </Box>
          <Box
            className={cn(
              'text-xxs',
              Opacity.READ,
              !settings.showNumber && 'hidden',
            )}
          >
            {notificationNumber}
          </Box>
        </Stack>
      </Box>
    )
  );
};
