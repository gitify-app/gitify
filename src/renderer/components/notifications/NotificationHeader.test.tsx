import { fireEvent, render, screen } from '@testing-library/react';
import { mockSettings } from '../../__mocks__/state-mocks';
import { AppContext } from '../../context/App';
import { GroupBy } from '../../types';
import { mockSingleNotification } from '../../utils/api/__mocks__/response-mocks';
import * as comms from '../../utils/comms';
import { NotificationHeader } from './NotificationHeader';

describe('renderer/components/notifications/NotificationHeader.tsx', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render itself & its children - group by repositories', async () => {
    const props = {
      notification: mockSingleNotification,
    };

    const tree = render(
      <AppContext.Provider
        value={{ settings: { ...mockSettings, groupBy: GroupBy.REPOSITORY } }}
      >
        <NotificationHeader {...props} />
      </AppContext.Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  describe('should render itself & its children - group by date', () => {
    it('with notification number', async () => {
      const props = {
        notification: mockSingleNotification,
      };

      const tree = render(
        <AppContext.Provider
          value={{ settings: { ...mockSettings, groupBy: GroupBy.DATE } }}
        >
          <NotificationHeader {...props} />
        </AppContext.Provider>,
      );
      expect(tree).toMatchSnapshot();
    });

    it('with showNumber setting disabled', async () => {
      const props = {
        notification: mockSingleNotification,
      };

      const tree = render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              showNumber: false,
              groupBy: GroupBy.DATE,
            },
          }}
        >
          <NotificationHeader {...props} />
        </AppContext.Provider>,
      );
      expect(tree).toMatchSnapshot();
    });

    it('without notification number', async () => {
      const props = {
        notification: {
          ...mockSingleNotification,
          subject: { ...mockSingleNotification.subject, number: null },
        },
      };

      const tree = render(
        <AppContext.Provider
          value={{ settings: { ...mockSettings, groupBy: GroupBy.DATE } }}
        >
          <NotificationHeader {...props} />
        </AppContext.Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });

  it('should open notification user profile - group by date', () => {
    const openExternalLinkMock = jest
      .spyOn(comms, 'openExternalLink')
      .mockImplementation();

    const props = {
      notification: mockSingleNotification,
    };

    render(
      <AppContext.Provider
        value={{ settings: { ...mockSettings, groupBy: GroupBy.DATE } }}
      >
        <NotificationHeader {...props} />
      </AppContext.Provider>,
    );

    fireEvent.click(screen.getByTestId('view-repository'));
    expect(openExternalLinkMock).toHaveBeenCalledTimes(1);
    expect(openExternalLinkMock).toHaveBeenCalledWith(
      props.notification.repository.html_url,
    );
  });
});
