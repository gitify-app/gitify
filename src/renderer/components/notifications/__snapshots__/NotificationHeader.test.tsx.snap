// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/notifications/NotificationHeader.tsx should render itself & its children - group by date with notification number 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-g0xbh4-0 py-0.5"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Box-sc-g0xbh4-0 text-xs font-medium"
            data-testid="view-repository"
            title="Open repository"
          >
            <div
              class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
              data-align="center"
              data-direction="horizontal"
              data-gap="condensed"
              data-justify="start"
              data-padding="none"
              data-testid="avatar"
              data-wrap="nowrap"
            >
              <img
                alt="gitify-app/notifications-test"
                class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
                data-component="Avatar"
                height="14"
                size="14"
                src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
                sx="[object Object]"
                width="14"
              />
              <div
                class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
                maxwidth="280"
                title="gitify-app/notifications-test"
              >
                gitify-app/notifications-test
              </div>
            </div>
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-xxs opacity-50"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-g0xbh4-0 py-0.5"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="center"
        data-direction="horizontal"
        data-gap="condensed"
        data-justify="start"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Box-sc-g0xbh4-0 text-xs font-medium"
          data-testid="view-repository"
          title="Open repository"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-testid="avatar"
            data-wrap="nowrap"
          >
            <img
              alt="gitify-app/notifications-test"
              class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
              data-component="Avatar"
              height="14"
              size="14"
              src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
              sx="[object Object]"
              width="14"
            />
            <div
              class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
              maxwidth="280"
              title="gitify-app/notifications-test"
            >
              gitify-app/notifications-test
            </div>
          </div>
        </div>
        <div
          class="Box-sc-g0xbh4-0 text-xxs opacity-50"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/NotificationHeader.tsx should render itself & its children - group by date with showNumber setting disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-g0xbh4-0 py-0.5"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Box-sc-g0xbh4-0 text-xs font-medium"
            data-testid="view-repository"
            title="Open repository"
          >
            <div
              class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
              data-align="center"
              data-direction="horizontal"
              data-gap="condensed"
              data-justify="start"
              data-padding="none"
              data-testid="avatar"
              data-wrap="nowrap"
            >
              <img
                alt="gitify-app/notifications-test"
                class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
                data-component="Avatar"
                height="14"
                size="14"
                src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
                sx="[object Object]"
                width="14"
              />
              <div
                class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
                maxwidth="280"
                title="gitify-app/notifications-test"
              >
                gitify-app/notifications-test
              </div>
            </div>
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-xxs opacity-50 hidden"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-g0xbh4-0 py-0.5"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="center"
        data-direction="horizontal"
        data-gap="condensed"
        data-justify="start"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Box-sc-g0xbh4-0 text-xs font-medium"
          data-testid="view-repository"
          title="Open repository"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-testid="avatar"
            data-wrap="nowrap"
          >
            <img
              alt="gitify-app/notifications-test"
              class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
              data-component="Avatar"
              height="14"
              size="14"
              src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
              sx="[object Object]"
              width="14"
            />
            <div
              class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
              maxwidth="280"
              title="gitify-app/notifications-test"
            >
              gitify-app/notifications-test
            </div>
          </div>
        </div>
        <div
          class="Box-sc-g0xbh4-0 text-xxs opacity-50 hidden"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/NotificationHeader.tsx should render itself & its children - group by date without notification number 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-g0xbh4-0 py-0.5"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Box-sc-g0xbh4-0 text-xs font-medium"
            data-testid="view-repository"
            title="Open repository"
          >
            <div
              class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
              data-align="center"
              data-direction="horizontal"
              data-gap="condensed"
              data-justify="start"
              data-padding="none"
              data-testid="avatar"
              data-wrap="nowrap"
            >
              <img
                alt="gitify-app/notifications-test"
                class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
                data-component="Avatar"
                height="14"
                size="14"
                src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
                sx="[object Object]"
                width="14"
              />
              <div
                class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
                maxwidth="280"
                title="gitify-app/notifications-test"
              >
                gitify-app/notifications-test
              </div>
            </div>
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-xxs opacity-50"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-g0xbh4-0 py-0.5"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="center"
        data-direction="horizontal"
        data-gap="condensed"
        data-justify="start"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Box-sc-g0xbh4-0 text-xs font-medium"
          data-testid="view-repository"
          title="Open repository"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-testid="avatar"
            data-wrap="nowrap"
          >
            <img
              alt="gitify-app/notifications-test"
              class="Avatar__StyledAvatar-sc-2lv0r8-0 htlvKA"
              data-component="Avatar"
              height="14"
              size="14"
              src="https://avatars.githubusercontent.com/u/133795385?s=200&v=4"
              sx="[object Object]"
              width="14"
            />
            <div
              class="Truncate__StyledTruncate-sc-23o1d2-0 ldPZNj"
              maxwidth="280"
              title="gitify-app/notifications-test"
            >
              gitify-app/notifications-test
            </div>
          </div>
        </div>
        <div
          class="Box-sc-g0xbh4-0 text-xxs opacity-50"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/NotificationHeader.tsx should render itself & its children - group by repositories 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
